subprojects/plugins/src/main/java/org/gradle/api/internal/plugins/DefaultArtifactPublicationSet.java

```diff
@@ -41,7 +42,12 @@ public DefaultArtifactPublicationSet(PublishArtifactSet artifactContainer) {
     public void addCandidate(PublishArtifact artifact) {
         if (defaultArtifactProvider == null) {
             defaultArtifactProvider = new DefaultArtifactProvider();
-            artifactContainer.addAllLater(defaultArtifactProvider);
+            MutationGuards.of(artifactContainer).whileMutationEnabled(new Runnable() {
```

Comment:
Why's this needed? A client of a container should _never_ reach in and disable the guard of the container.

Reply:
I thought this is what we discussed that Gradle is privileged and can go and disable the guard if he knows what he is doing. It certainly not what users should be doing. I fixed the `BasePlugin` to skip over the `PublishArtifactSet` that cause an issue with this code. I hope it will be alright as I did that fix previously and it was failing somewhere else relying on some ordering.